<?xml version="1.6" ?>
<launch>

	<!--> Node Name <-->
    <arg name="control_node_name" default="default_central"/>


    <!--> ROS Topic Name <-->
	<arg name="subscribe_topic_back" default="/default/frontLines"/>
	<arg name="subscribe_topic_front" default="/default/backLines"/>

	<arg name="publish_topic_left" default="/default/leftWheel"/>
	<arg name="publish_topic_right" default="/default/rightWheel"/>

    <arg name="change_mode_topic" default="none"/>
    <arg name="emergency_topic" default="none"/>

    <arg name="rviz_topic" default="/default/selectedLines"/>


	<!--> Robot Infos <-->
	<arg name="mode" default="automatic"/>

	<arg name="number_lines" default="4"/>
	<arg name="turn_times" default="2"/>
	<arg name="sleep_time_ms" default="250"/>

	<arg name="max_velocity" default="1.0"/>
	<arg name="body_size" default="1.0"/>

	<arg name="rviz_frame" default="world"/>


	<!--> Launch node <-->
	<node name="$(arg control_node_name)" pkg="engrais_control" type="robot_control" output="screen">

		<param name="subscribe_topic_back" type="string" value="$(arg subscribe_topic_back)"/>
		<param name="subscribe_topic_front" type="string" value="$(arg subscribe_topic_front)"/>

		<param name="publish_topic_left" type="string" value="$(arg publish_topic_left)"/>
		<param name="publish_topic_right" type="string" value="$(arg publish_topic_right)"/>

        <param name="mode" type="string" value="$(arg mode)"/>
        <param name="change_mode_topic" type="string" value="$(arg change_mode_topic)"/>

		<param name="number_lines" type="int" value="$(arg number_lines)"/>
		<param name="turn_times" type="int" value="$(arg turn_times)"/>
		<param name="sleep_time_ms" type="int" value="$(arg sleep_time_ms)"/>

		<param name="max_velocity" type="double" value="$(arg max_velocity)"/>
		<param name="body_size" type="double" value="$(arg body_size)"/>

		<param name="rviz_frame" type="string" value="$(arg rviz_frame)"/>
		<param name="rviz_topic" type="string" value="$(arg rviz_topic)"/>
        <param name="emergency_topic" type="string" value="$(arg emergency_topic)"/>

	</node>

</launch>